libraries/chain/block_schedule.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright (c) 2017, Respective Authors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <eos/chain/block_schedule.hpp>
+#include <eos/chain/block.hpp>
+
+namespace eos { namespace chain {
+
+static uint next_power_of_two(uint input) {
+   if (input == 0) {
+      return 0;
+   }
+
+   uint result = input;
+   result--;
+   result |= result >> 1;
+   result |= result >> 2;
+   result |= result >> 4;
+   result |= result >> 8;
+   result |= result >> 16;
+   result++;
+   return result;
+};
+
+typedef std::hash<decltype(AccountName::value)> account_hash;
+static account_hash account_hasher;
+
+struct schedule_entry {
+   schedule_entry(uint _cycle, uint _thread, pending_transaction const * _transaction) 
+      : cycle(_cycle)
+      , thread(_thread)
+      , transaction(_transaction) 
+   {}
+
+   uint cycle;
+   uint thread;
+   pending_transaction const *transaction;
```

Comment:
stylistically, the * belongs to the type not the variable.  Also, since transaction cannot be null it should be a const & and no pointers should be used at all.